src/RedisEnterpriseCache/custom/Invoke-AzRedisEnterpriseCacheForceDatabaseUnlink.ps1

```diff
@@ -0,0 +1,172 @@
+    ${Parameter},
+
+    [Parameter(ParameterSetName='ForceExpanded', Mandatory)]
+    [Parameter(ParameterSetName='ForceViaIdentityExpanded', Mandatory)]
+    [Microsoft.Azure.PowerShell.Cmdlets.RedisEnterpriseCache.Category('Body')]
+    [System.String[]]
+    # The resource IDs of the database resources to be unlinked.
+    ${Id},
+
+    [Parameter()]
+    [Alias('AzureRMContext', 'AzureCredential')]
+    [ValidateNotNull()]
+    [Microsoft.Azure.PowerShell.Cmdlets.RedisEnterpriseCache.Category('Azure')]
+    [System.Management.Automation.PSObject]
+    # The credentials, account, tenant, and subscription used for communication with Azure.
+    ${DefaultProfile},
+
+    [Parameter()]
+    [Microsoft.Azure.PowerShell.Cmdlets.RedisEnterpriseCache.Category('Runtime')]
+    [System.Management.Automation.SwitchParameter]
+    # Run the command as a job
+    ${AsJob},
+
+    [Parameter(DontShow)]
+    [Microsoft.Azure.PowerShell.Cmdlets.RedisEnterpriseCache.Category('Runtime')]
+    [System.Management.Automation.SwitchParameter]
+    # Wait for .NET debugger to attach
+    ${Break},
+
+    [Parameter(DontShow)]
+    [ValidateNotNull()]
+    [Microsoft.Azure.PowerShell.Cmdlets.RedisEnterpriseCache.Category('Runtime')]
+    [Microsoft.Azure.PowerShell.Cmdlets.RedisEnterpriseCache.Runtime.SendAsyncStep[]]
+    # SendAsync Pipeline Steps to be appended to the front of the pipeline
+    ${HttpPipelineAppend},
+
+    [Parameter(DontShow)]
+    [ValidateNotNull()]
+    [Microsoft.Azure.PowerShell.Cmdlets.RedisEnterpriseCache.Category('Runtime')]
+    [Microsoft.Azure.PowerShell.Cmdlets.RedisEnterpriseCache.Runtime.SendAsyncStep[]]
+    # SendAsync Pipeline Steps to be prepended to the front of the pipeline
+    ${HttpPipelinePrepend},
+
+    [Parameter()]
+    [Microsoft.Azure.PowerShell.Cmdlets.RedisEnterpriseCache.Category('Runtime')]
+    [System.Management.Automation.SwitchParameter]
+    # Run the command asynchronously
+    ${NoWait},
+
+    [Parameter()]
+    [Microsoft.Azure.PowerShell.Cmdlets.RedisEnterpriseCache.Category('Runtime')]
+    [System.Management.Automation.SwitchParameter]
+    # Returns true when the command succeeds
+    ${PassThru},
+
+    [Parameter(DontShow)]
+    [Microsoft.Azure.PowerShell.Cmdlets.RedisEnterpriseCache.Category('Runtime')]
+    [System.Uri]
+    # The URI for the proxy server to use
+    ${Proxy},
+
+    [Parameter(DontShow)]
+    [ValidateNotNull()]
+    [Microsoft.Azure.PowerShell.Cmdlets.RedisEnterpriseCache.Category('Runtime')]
+    [System.Management.Automation.PSCredential]
+    # Credentials for a proxy server to use for the remote call
+    ${ProxyCredential},
+
+    [Parameter(DontShow)]
+    [Microsoft.Azure.PowerShell.Cmdlets.RedisEnterpriseCache.Category('Runtime')]
+    [System.Management.Automation.SwitchParameter]
+    # Use the default credentials for the proxy
+    ${ProxyUseDefaultCredentials}
+)
+
+    process {
+        $null = $PSBoundParameters.Add("DatabaseName", "default")
```

Comment:
The easiest way to add default value is by directive, see https://github.com/Azure/autorest.powershell/blob/main/docs/directives.md#default-values. And your change will make DatabaseName invisible to customer. Is it expected?  If so, you can hide parameter by directive without customization as well, see https://github.com/Azure/autorest.powershell/blob/main/docs/directives.md#parameter-hiding

Reply:
`DatabaseName` is supposed to be a compulsory parameter which is hidden from the customer, and we always provide the value `default` for it. I did ask the team whether this was possible and was suggested to write a custom cmdlet instead. This is a reoccurring pattern used in many of our custom cmdlets.

https://github.com/Azure/autorest.powershell/blob/main/docs/directives.md#default-values mentions that once a default value script is declared for a parameter, that parameter will be made optional at build-time, which isn't something we want.

Reply:
Yes. Giving it a default value will provide a value for the parameter in all time. That's why It will be marked as optional in command syntax. But in c# sharp implementation, it is still required. See example here https://github.com/Azure/azure-powershell/blob/086ce98d157d5fe3513d3a4c6dc1ff9412fd06cf/src/Logz/README.md?plain=1#L135-L143, https://github.com/Azure/azure-powershell/blob/086ce98d157d5fe3513d3a4c6dc1ff9412fd06cf/src/Logz/generated/cmdlets/NewAzLogzMonitorTagRule_CreateExpanded.cs#L174

Reply:
Added default value for DatabaseName parameter to README.md, and removed customized force unlink cmdlet.